core/modules/qdisp/QueryRequest.cc

```diff
@@ -54,18 +54,137 @@ namespace lsst {
 namespace qserv {
 namespace qdisp {
 
+
+// Run action() when the system expects to have time to accept data.
+class QueryRequest::AskForResponseDataCmd : public PriorityCommand {
+public:
+    typedef std::shared_ptr<AskForResponseDataCmd> Ptr;
+    enum class State { STARTED0, DATAREADY1, DONE2 };
+    AskForResponseDataCmd(QueryRequest::Ptr const& qr, JobQuery::Ptr const& jq)
+        : _qRequest(qr), _jQuery(jq), _idStr(jq->getIdStr()) {}
+
+    void action(util::CmdData *data) override {
+        // If everything is ok, call GetResponseData to have XrdSsi ask the worker for the data.
+        {
+            auto jq = _jQuery.lock();
+            auto qr = _qRequest.lock();
+            if (jq == nullptr || qr == nullptr) {
+                LOGS(_log, LOG_LVL_WARN, _idStr << " AskForResponseData null before GetResponseData");
+                // No way to call _errorFinish().
+                _setState(State::DONE2);
+                return;
+            }
+
+            if (qr->isQueryCancelled()) {
+                LOGS(_log, LOG_LVL_DEBUG, _idStr << " AskForResponseData query was cancelled");
+                qr->_errorFinish(true);
+                _setState(State::DONE2);
+                return;
+            }
+            std::vector<char>& buffer = jq->getDescription()->respHandler()->nextBuffer();
+            LOGS(_log, LOG_LVL_DEBUG, _idStr << " Asking for GetResponseData size=" << buffer.size());
+            qr->GetResponseData(&buffer[0], buffer.size());
```

Comment:
In general I prefer more readable `buffer.data()` to `&buffer[0]`, but this is clearly personal preference 🐱